pkg/health/checker/checker.go

```diff
@@ -0,0 +1,202 @@
+package checker
```

Comment:
It's a cherry-pick of PR #238. Review comments should go there. (Preview: I only moved around a bunch of functions.)